src/App.tsx

```diff
@@ -1,204 +1,112 @@
+        setErrorMessage('Unable to load todos');
+        setShowError(true);
+        setTimeout(() => {
+          setShowError(false);
+        }, 3000);
+      }
+    };
+
+    loadData();
+
+    // CLEEARING ERRORS before next API call
+    // not sure if this is correct
+    return () => {
+      setShowError(false);
+      setErrorMessage('');
+    };
+  }, []);
 
-const USER_ID = 0;
-
-export const App: React.FC = () => {
   if (!USER_ID) {
     return <UserWarning />;
   }
 
+  // FILTERING TODOS
+  let filteredTodos;
+
+  if (todos?.length) {
+    filteredTodos = [...todos];
+    switch (selectedFilter) {
+      case 'Active':
+        filteredTodos = filteredTodos.filter(todo => !todo.completed);
+        break;
+      case 'Completed':
+        filteredTodos = filteredTodos.filter(todo => todo.completed);
+        break;
+      default:
+        break;
+    }
+  }
+
   return (
     <div className="todoapp">
       <h1 className="todoapp__title">todos</h1>
 
       <div className="todoapp__content">
-        <header className="todoapp__header">
-          {/* this buttons is active only if there are some active todos */}
-          <button
-            type="button"
-            className="todoapp__toggle-all active"
-            data-cy="ToggleAllButton"
-          />
-
-          {/* Add a todo on form submit */}
-          <form>
-            <input
-              data-cy="NewTodoField"
-              type="text"
-              className="todoapp__new-todo"
-              placeholder="What needs to be done?"
-            />
-          </form>
-        </header>
-
-        <section className="todoapp__main" data-cy="TodoList">
-          {/* This is a completed todo */}
-          <div data-cy="Todo" className="todo completed">
-            <label className="todo__status-label">
-              <input
-                data-cy="TodoStatus"
-                type="checkbox"
-                className="todo__status"
-                checked
-              />
-            </label>
-
-            <span data-cy="TodoTitle" className="todo__title">
-              Completed Todo
-            </span>
-
-            {/* Remove button appears only on hover */}
-            <button type="button" className="todo__remove" data-cy="TodoDelete">
-              ×
-            </button>
-
-            {/* overlay will cover the todo while it is being updated */}
-            <div data-cy="TodoLoader" className="modal overlay">
-              <div className="modal-background has-background-white-ter" />
-              <div className="loader" />
-            </div>
-          </div>
-
-          {/* This todo is not completed */}
-          <div data-cy="Todo" className="todo">
-            <label className="todo__status-label">
-              <input
-                data-cy="TodoStatus"
-                type="checkbox"
-                className="todo__status"
+        <Header />
+        {
+          todos && !!todos.length && (
```

Comment:
In my opinion that is the most readable
```suggestion
          todos.length > 0  (
```